### PR TITLE
fix(toolchain): expose onedir command-dependency tools on step PATH

### PR DESCRIPTION
**Pull request by sgtoj**
## what

- Fix onedir (multi-file) tools declared in a custom command's `dependencies.tools` (e.g. `nodejs/node`, `npm/cli`, `aws-cli`) not being reachable from the command's `steps` — they previously failed with `executable file not found in $PATH` (exit 127).
- `BuildToolchainPATH` now resolves each dependency's PATH entries through the same onedir-aware logic as `atmos toolchain env`, adding the nested `.pkg/.../bin` directory(ies) recorded in `.atmos-onedir.json` instead of the bare version directory.
- Add `toolchain.EntrypointDirsForVersion` as the single source of truth for onedir entrypoint-directory resolution, shared by `atmos toolchain env`, the dependency PATH builder (custom commands, hooks, terraform/helmfile/packer, ansible), and `ToolchainEnvironment`.
- Make `ToolchainEnvironment` dirs (`ToolchainDirs()` / `PrependToPath()`) onedir-complete — every entrypoint directory, not just the primary binary's.
- Flat single-binary tools are unchanged (still their version dir); a not-installed tool falls back to the bare version dir (preserves existing behavior).

## why

- Onedir installs (#2750) keep executables nested under `.pkg/.../bin` per `.atmos-onedir.json`, but the command-dependency PATH builder still prepended the bare version dir, so node/npm/aws-cli were never on PATH for steps.
- `atmos toolchain env` already resolved these correctly; the command-dependency path simply wasn't reusing that resolution. Unifying both behind one helper makes onedir tools generally usable as command/hook/component dependencies.
- This is the remaining gap after #2750 (onedir installs) and #2754 (onedir symlink materialization); this change is solely about how onedir command-dependency tools are exposed on PATH.

## references

- Builds on #2750 (multi-file "onedir" installs) and #2754 (onedir symlink entrypoints).
- Repro: a custom command with `dependencies.tools: { nodejs/node: v24.18.0 }` whose step runs `node --version` now succeeds; a single-binary control (`kubectl`) keeps resolving.

<details><summary>Testing</summary>

- New unit tests, with all net-new code covered (`EntrypointDirsForVersion` 100%, `BuildToolchainPATH` 100%, `collectToolchainDirs` 94.7%, `addEntrypointDirs` 100%):
  - `TestEntrypointDirsForVersion_*` (pkg/toolchain): onedir multi-dir, flat, not-installed.
  - `TestBuildToolchainPATH_Onedir` (pkg/dependencies): asserts `exec.LookPath` resolves the nested onedir entrypoint, flat `kubectl` still resolves, and a shared onedir dir is deduplicated. Verified this **fails on pre-fix code** and passes after.
  - `TestNewEnvironment_OnedirDirs` (pkg/dependencies): `env.dirs` onedir-complete, dedup + absolutize, primary-dir fallback.
- `go build ./...`, `go vet`, custom `golangci-lint` (patch-scoped), and `gofumpt` are clean; consumer packages (hooks, ansible, helm, terraform/output) pass.
- Cross-platform: `.exe` handling + `filepath.Join`. A Windows acceptance subset for `./pkg/toolchain/...` and `./pkg/dependencies/...` passed, including `TestBuildToolchainPATH_Onedir`, `TestEntrypointDirsForVersion_*`, and `TestNewEnvironment_OnedirDirs`.

</details>

## Summary by CodeRabbit

* **New Features**
  * Improved tool discovery for onedir-style installations.
  * Automatically prepends required entrypoint directories to the toolchain `PATH`, deduplicated and normalized (absolute, order preserved).
  * Preserves flat-install behavior by falling back to the tool’s primary binary directory.
* **Bug Fixes**
  * Ensures executable resolution from onedir layouts works reliably; returns `PATH` unchanged when no resolved directories are available (including for empty `PATH` input).
* **Tests**
  * Added coverage for onedir entrypoint directory behavior and `BuildToolchainPATH` edge cases (empty `PATH` handling).

### CODE REVIEW

File: pkg/dependencies/installer.go [hunk 1]
```diff
@@ -566,38 +574,62 @@ func BuildToolchainPATH(atmosConfig *schema.AtmosConfiguration, dependencies map
...
+	paths := collectToolchainDirs(locator, resolver, binDir, dependencies)
+
+	// Prepend toolchain paths to existing PATH.
+	currentPath := getPathFromEnv()
+	if len(paths) == 0 {
+		return currentPath, nil
+	}
+
+	newPath := strings.Join(append(paths, currentPath), string(os.PathListSeparator))
+	return newPath, nil
```
[the remark this answers is not in the document]
Comment (sgtoj) [reply pkg/dependencies/installer.go]: Good catch — fixed in 3f5842d7c. `BuildToolchainPATH` now appends the current PATH only when it's non-empty, so an unset/empty PATH no longer leaves a trailing list separator (which exec lookups treat as `.`). Added a regression test `TestBuildToolchainPATH_EmptyPATHNoTrailingSeparator` asserting no trailing separator and no empty component.

File: pkg/toolchain/path_helpers.go [hunk 2]
```diff
@@ -61,6 +62,29 @@ func entrypointDirs(binaryPaths []string, relativeFlag bool) []string {
...
+// as `atmos toolchain env`. For a onedir (multi-file) package it returns each
+// directory holding a resolved manifest entrypoint (e.g. the nested .pkg/.../bin);
+// for a flat install it returns the single version directory. It returns one entry
+// per resolved entrypoint (callers deduplicate), and nil when the tool version is
+// not installed, letting the caller fall back to a bare version-dir guess for
+// backward compatibility.
+//
+// When relativeFlag is true the directories are relative to the locator's bin
+// dir; otherwise they are absolute (the common case for building a subprocess
+// PATH).
```
[the remark this answers is not in the document]
Comment (sgtoj) [reply pkg/toolchain/path_helpers.go]: Fixed the doc in 3f5842d7c. Updated the `relativeFlag` godoc on `EntrypointDirsForVersion` to state that when true the locator-provided path form is preserved (paths are not absolutized, so they stay relative only when the locator's bin dir is relative), and when false the directories are absolutized. This matches the implementation (`resolveDirPath`) and the tests.
